### PR TITLE
Fix bug where highways could be built on invalid terrain.

### PR DESCRIPTION
**Pull request by PrettyFlower**

### CODE REVIEW

File: src/building/construction.c [hunk 1]
```diff
@@ -662,9 +663,7 @@ void building_construction_cancel(void)
 {
     map_property_clear_constructing_and_deleted();
     if (data.in_progress && building_construction_is_updatable()) {
-        if (building_construction_is_updatable()) {
-            game_undo_restore_map(1);
-        }
+        game_undo_restore_map(1);
```
Comment (crudelios) [inline src/building/construction.c:666]: Why this change?

Comment (PrettyFlower) [reply src/building/construction.c:666]: building_construction_is_updatable() has to be true in order to make it into this if block, so checking here again is redundant.
